tasks/transform_conformance/tests/legacy-decorators/test/fixtures/index.js

```diff
@@ -0,0 +1,41 @@
+import {  } from 'fs';
+import { transpileModule } from 'typescript'
+import { readdir, writeFile, rename, mkdir, readFile } from 'fs/promises';
+import { basename, dirname, join } from 'path';
+const __dirname = new URL('.', import.meta.url).pathname + 'typescript';
+
+
+// async function rename() {
+//   const files = await readdir(__dirname, {recursive: true});
+//   files.map((file) => join(__dirname, file) ).forEach(async file => {
+//     if (file.endsWith(".ts")) {
+//       const newFile = file.replace(/\.ts$/, '/input.ts');
+//       await mkdir(file.replace(/\.ts$/, ""), {recursive: true});
+//       await rename(file, newFile);
+//     }
+//   });
+// }
+async function main() {
+  const files = await readdir(__dirname, {recursive: true});
+  files.map((file) => join(__dirname, file) ).forEach(async file => {
```

Comment:
This is very Rust-y Javascript!

Personally, I'd just do it with a `for` loop (run in series) because it'd be simpler to understand, and reading/writing 70 files simultaneously probably isn't very efficient anyway - it probably overloads the file system.

But if you want to keep it async, you should `await` the resulting promises, or they're just "firing off into space", and the function returns before they're all done. I'm not sure if errors would get handled.

```js
await Promise.all(
  files.map((file) => join(__dirname, file) )
    // NOTE: `.map` not `.forEach`
    .map(async file => {
      // Do async stuff
    })
);
```

I guess it doesn't matter much since all the results are checked into git, but since it's committed to the repo...

Reply:
Oh, I mistakenly committed it, I initially thought that would never use it again, so writing it casually. But you said right, it would be better if we can retrace in the future. I will re-implement it.